api/task_manager.cc

```diff
@@ -0,0 +1,202 @@
+#include <seastar/core/coroutine.hh>
+
+#include "task_manager.hh"
+#include "api/api-doc/task_manager.json.hh"
+#include "db/system_keyspace.hh"
+#include "column_family.hh"
+#include "unimplemented.hh"
+#include "storage_service.hh"
+
+#include <utility>
+
+namespace api {
+
+namespace tm = httpd::task_manager_json;
+using namespace json;
+
+inline bool filter_tasks(seastar::shared_ptr<task_manager::task> task, std::unordered_map<sstring, sstring>& query_params) {
+    return (!query_params.contains("keyspace") || query_params["keyspace"] == task->get_status().keyspace) &&
+        (!query_params.contains("table") || query_params["table"] == task->get_status().table);
+}
+
+struct status {
+    task_manager::task_status task_status;
+    task_manager::running_tasks number;
+    task_manager::module module;
+    unsigned shard;
+};
+
+tm::task_status make_status(status status) {
+    auto start_time = gc_clock::to_time_t(status.task_status.start_time);
+    auto end_time = gc_clock::to_time_t(status.task_status.end_time);
+    ::tm st, et;
+    ::gmtime_r(&end_time, &et);
+    ::gmtime_r(&start_time, &st);
+
+    tm::task_status res{};
+    res.id = status.task_status.id.to_sstring();
+    res.type = status.task_status.type;
+    res.state = status.task_status.state;
```

Comment:
don't we need to convert the enum to string?

Reply:
It is defined as enum in task_manager.json (https://github.com/scylladb/scylladb/blob/33efb27a69e9246ef7482561d7d1d2ad849ac634/api/api-doc/task_manager.json#L181-L190)
and thus a type with corresponding fields is required.

Reply:
ok